cmd/docker/docker.go

```diff
@@ -274,6 +302,22 @@ func runDocker(dockerCli *command.DockerCli) error {
 		return err
 	}
 
+	// Buildkit's detect package currently follows the old otel spec which defaulted to gRPC.
+	// Since the spec changed to default to http/protobuf.
+	// If these env vars are not set then we set them to the new default so detect will give us the expected protocol.
+	// This is the same as on the dockerd side.
+	// This can be removed after buildkit's detect package is updated.
+	if os.Getenv("OTEL_EXPORTER_OTLP_TRACES_PROTOCOL") == "" && os.Getenv("OTEL_EXPORTER_OTLP_PROTOCOL") == "" {
+		os.Setenv("OTEL_EXPORTER_OTLP_TRACES_PROTOCOL", "http/protobuf")
```

Comment:
This probably is not needed since we aren't using detect anymore?

Reply:
Yeah, it should go away - I verified things are still good without it

(@thaJeztah I deleted it over in https://github.com/thaJeztah/cli/pull/4/files#diff-3dcd6325b70a5f5db393a3ecc08060b69007fcef371fb4dabf3164da95646fb6L305-L312 )